controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -592,6 +596,128 @@ func (r *BareMetalHostReconciler) detachHost(prov provisioner.Provisioner, info
 	return slowPoll
 }
 
+type imageBuildError struct {
+	Message string
+}
+
+func (ibe imageBuildError) Error() string {
+	return ibe.Message
+}
+
+func (r *BareMetalHostReconciler) preprovImageCurrent(info *reconcileInfo, image *metal3v1alpha1.PreprovisioningImage) (bool, error) {
```

Comment:
naming: what about something like `preprovImageReady` to make it a little bit more clear?